### PR TITLE
feat: add stage backend publish task

### PR DESCRIPTION
**Pull request by zhu327**
### Description

Fixes # (issue)

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/dashboard/apigateway/apigateway/apis/open/stage/serializers.py [hunk 1]
```diff
@@ -42,3 +65,189 @@ def to_representation(self, instance):
...
+    def validate(self, data):
+        if data.get("loadbalance") == LoadBalanceTypeEnum.WRR.value:
+            host_without_weight = [host for host in data["hosts"] if host.get("weight") is None]
+            if host_without_weight:
+                raise serializers.ValidationError(_("负载均衡类型为 Weighted-RR 时，Host 权重必填。"))
+        return data
+
+
+class TransformHeadersSLZ(serializers.Serializer):
+    # add = serializers.DictField(child=serializers.CharField(), required=False, allow_empty=True)
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/apis/open/stage/serializers.py:114]: 没用的代码删了

File: src/dashboard/apigateway/apigateway/apis/web/backend/views.py [hunk 2]
```diff
@@ -138,7 +145,9 @@ def update(self, request, *args, **kwargs):
     def destroy(self, request, *args, **kwargs):
         instance = self.get_object()
 
-        # TODO 通过stage/resource关联数据校验是否能删除
+        # 通过stage/resource关联数据校验是否能删除
+        if not BackendHandler.deletable(instance):
+            raise error_codes.INVALID_ARGUMENT.format(_("请先下线后端服务，然后再删除。"))
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/apis/web/backend/views.py:150]: error_codes.FAILED_PRECONDITION

**Review by wklken — COMMENTED** [review]
LGTM

File: src/dashboard/apigateway/apigateway/apis/web/backend/views.py (hunk 2 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/apis/web/backend/views.py]: Done

**Review by wklken — APPROVED** [review]

**Pull request merged**